### PR TITLE
[Chore] Build and publish Telemetrygen images with multiarch support

### LINKED ISSUE #30072: Request for Multiarch Support in Telemetrygen Container Image.
### Component(s)

cmd/telemetrygen

### Is your feature request related to a problem? Please describe.

The OpenTelemetry Operator and Collector are compatible with multiple architectures including linux/amd64, linux/arm64, linux/ppc64le, and linux/s390x. However, the E2E tests in the operator repository rely on Telemetrygen, which currently lacks multiarch support.

To enable testing on all supported platforms, it’s crucial that the Telemetrygen image is built with multiarch support. This enhancement will allow us to run our E2E tests across all supported architectures, ensuring comprehensive coverage and compatibility.

We propose that the Telemetrygen image be built directly from the project repository, which would streamline the process and maintain consistency. This change would be a significant improvement for our testing infrastructure.

### Describe the solution you'd like

Update the Telemetrygen [workflow](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/telemetrygen.yml) to build and publish multiarch images.  

### Describe alternatives you've considered

Build the multiarch images from the forked repo and publish it to a container registry. Replace the image in tests. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by IshwarKanse**
**Description:** <Describe what has changed.>
This PR updates the Telemetrygen workflow to build and publish multiarch images. 

Resolves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30072

**Testing:** <Describe what testing was performed and which tests were added.>

ToDo

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by bryan-aguilar** [comment]
Hi @IshwarKanse,

Is there a tracking issue where this change has been discussed among the collector community already?

**Comment by IshwarKanse** [comment]
@bryan-aguilar  I didn't find a existing issue so created one for this change. https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30072

**Comment by bryan-aguilar** [comment]
Thank you, I was just curious to know if I was missing any previous context. I think discussion can be had on the PR rather than the issue. I think the issue summarizes up the problem well and I personally don't see immediate issues with this. 

@open-telemetry/collector-maintainers group should probably have input on this also as it has to deal with an enhancement of release artifacts on this repository.

**Comment by bryan-aguilar** [comment]
This could be a good topic to ask at our weekly collector sigs that happen at 9AM PST every Wednesday. If you cannot make it this week I can bring this PR up for attention.

**Review by mx-psi — APPROVED** [review]

**Review by bryan-aguilar — APPROVED** [review]

File: .github/workflows/telemetrygen.yml [hunk 1]
```diff
@@ -51,6 +61,10 @@ jobs:
       packages: write
     steps:
       - uses: actions/checkout@v4
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```
Comment (bryan-aguilar) [inline .github/workflows/telemetrygen.yml:67]:
```suggestion
        uses: docker/setup-buildx-action@v3
```

File: .github/workflows/telemetrygen.yml [hunk 2]
```diff
@@ -51,6 +61,10 @@ jobs:
       packages: write
     steps:
       - uses: actions/checkout@v4
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
```
Comment (bryan-aguilar) [inline .github/workflows/telemetrygen.yml:65]:
```suggestion
        uses: docker/setup-qemu-action@v3
```

File: .github/workflows/telemetrygen.yml [hunk 3]
```diff
@@ -31,6 +36,10 @@ jobs:
...
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```
Comment (bryan-aguilar) [inline .github/workflows/telemetrygen.yml:42]:
```suggestion
        uses: docker/setup-buildx-action@v3
```

File: .github/workflows/telemetrygen.yml [hunk 4]
```diff
@@ -31,6 +36,10 @@ jobs:
       packages: write
     steps:
       - uses: actions/checkout@v4
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
```
Comment (bryan-aguilar) [inline .github/workflows/telemetrygen.yml:40]:
```suggestion
        uses: docker/setup-qemu-action@v3
```

File: .github/workflows/telemetrygen.yml [hunk 5]
```diff
@@ -17,12 +17,17 @@ jobs:
     if: ${{ github.actor != 'dependabot[bot]' && github.actor != 'renovate[bot]' }}
     steps:
       - uses: actions/checkout@v4
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```
Comment (bryan-aguilar) [inline .github/workflows/telemetrygen.yml:23]:
```suggestion
        uses: docker/setup-buildx-action@v3
```

File: .github/workflows/telemetrygen.yml [hunk 6]
```diff
@@ -17,12 +17,17 @@ jobs:
     if: ${{ github.actor != 'dependabot[bot]' && github.actor != 'renovate[bot]' }}
     steps:
       - uses: actions/checkout@v4
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
```
Comment (bryan-aguilar) [inline .github/workflows/telemetrygen.yml:21]:
```suggestion
        uses: docker/setup-qemu-action@v3
```

**Pull request merged**